cylc/flow/job_file.py

```diff
@@ -150,6 +152,22 @@ def _write_directives(self, handle, job_conf):
             for line in lines:
                 handle.write('\n' + line)
 
+    def _write_reinvocation(self, handle):
+        """Re-invoke using user determined bash interpreter."""
+        # NOTE this must be done after the directives are written out
+        # due to the way slurm reads directives
+        # NOTE we cannot do /usr/bin/env bash because we need to use the -l
+        # option and GNU env doesn't support additional arguments (recent
+        # versions permit this with the -S option similar to BSD env but we
+        # cannot make the jump to this until is it more widely adopted)
+        handle.write(dedent('''
+            if [[ $1 == 'noreinvoke' ]]; then
+                shift
+            else
+                exec "$(command -v bash)" -l "$0" noreinvoke "$@"
```

Comment:
Why do we need `exec "$(command -v bash)"`? How does this differ from just doing `exec bash`?

Reply:
Fair point.

Reply:
I've tested with `exec bash` and it seems to work fine

Reply:
Ok, I can't think of any reason not to go with that.

Reply:
I'm slightly worried that you wouldn't have used something as bizarre as `command -v bash` if there wasn't a good reason to do it!

But `command` is a bash built-in that by-passes function look-up, so would save us from users who define a shell function called `bash`? ... which does seem an unlikely problem to worry about.